Sources/PaymentsCore/Networking/HTTP.swift

```diff
@@ -0,0 +1,39 @@
+import Foundation
+
+/// `HTTP` interfaces directly with `URLSession` to execute network requests.
+class HTTP {
+    
+    private let coreConfig: CoreConfig
+    private var urlSession: URLSessionProtocol
+    private let decoder = APIClientDecoder()
+
+    init(coreConfig: CoreConfig) {
+        self.urlSession = URLSession.shared
+        self.coreConfig = coreConfig
+    }
+
+    /// For internal use for testing purpose
+    init(urlSession: URLSessionProtocol, coreConfig: CoreConfig) {
```

Comment:
Instead of having 2 init's since both are internal could we do:
```
    init(coreConfig: CoreConfig, urlSession: URLSessionProtocol = URLSession.shared)
```

Reply:
Good catch!